canonical-web-and-design/maas-ui
# Add processing labels to KVMList action forms.

## Done

- Added statuses to pod state.
- Added processing labels to KVMList action forms.
- Converted `useMachinesProcessing` hook into generic `useProcessing` hook, and used in KVMList forms.
- Rewrote pod reducer to allow for updating pod statuses (I'll try to refactor the `createStandardReducer` util to do this in #1279)

## QA

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Karura](/HACKING.md#karura)
- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- Go to /MAAS/r/kvm
- Select a few KVMs and choose Refresh from the action dropdown
- Check that a processing label shows how many KVMs have been refreshed
- Do the same but with deleting KVMs
- Check a couple of actions in the machine list as well, to make sure I haven't accidentally broken anything

## Fixes

Fixes #1300
